### DISCUSSION TITLE
How to return a status code other than 200 via Aggregation process?

### DISCUSSION THREAD

**Discussion by ggoyal-Axum** [ThreeMammals/Ocelot#1997] [Q&A] [answered]
Hi,
By default ocelot will return 200 even if any of the down streams returns 404, 401 etc during aggregation process.
**Is there a way to customize that behavior?**

Our UI-client will make decision based on returned API status code.
for example, if the status code is 401 then show a unauthorized screen
or if the status code is 400 then may be a different screen to show the issue and so on.

I have looked into custom `IDefinedAggregator` to achieve this for a specific route but I want the ability to change the default behavior for all the routes.

Currently i looking at replacing `SimpleJsonResponseAggregator` with custom but would like to know,
**Is there any better solution?**

**Comment by raman-m** [comment]
Hello, @ggoyal-Axum !
Welcome to Ocelot world! :tiger: 

> By default ocelot will return 200 even if any of the down streams returns 404, 401 etc during aggregation process. Is there a way to customize that behaviour?

Based on current implementation if downstream failed then the response will not be included in final aggregated data.
So, we will not change this behavior.

To customize route behavior for aggregated routes you need to develop custom `IDefinedAggregator`.
Then overriding status code should be possible.

**Comment by raman-m** [comment]
> I have looked into custom DefinedAggregator to achieve this for a specific route but I want the ability to change the default behaviour for all the routes.

For all routes you need to develop custom [Delegating Handler](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html) in which you override status code.

**Comment by raman-m** [comment, marked as answer]
> Currently i looking at replacing SimpleJsonResponseAggregator with custom but would like to know if there is any better solution

### I don't see your code :exclamation: 

For `Ocelot` package user, there is no better solution except: `IDefinedAggregator` and [Delegating Handler](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html)
For C# developer, there is better solution
- fork Ocelot repo into your GitHub account
- open in Visual Studio and develop whatever you want!
